### PR TITLE
feat(copilot): add single-agent autonomous skill (v01)

### PR DESCRIPTION
**Pull request by PengyuChen01**
## Summary
- Adds a GitHub Copilot–compatible variant of `autonomous-skill` under `copilot/skills/autonomous-copilot/`
- Runs the whole scan → score → fix → verify loop in **one agent session** (Copilot has no subagent dispatch, so the conductor → sprint master → worker architecture doesn't apply)
- Fully isolated from the Claude Code path: `copilot/` is never read by the existing root `SKILL.md` or any script

## Why
Copilot users currently can't consume `autonomous-skill` at all — the root `SKILL.md` dispatches subagents via `claude -p` + tmux, which Copilot doesn't support. This adds a parallel artifact that mirrors Claude skills' drop-in DX (single `SKILL.md` in a folder), discoverable by Copilot via the standard Agent Skill paths.

## Layout
```
copilot/
├── README.md                                  # 3 install paths
└── skills/
    └── autonomous-copilot/
        └── SKILL.md                           # Phase 1-4 workflow (Discovery → Scan → Fix → Summary)
```

## Install (for Copilot users)
| Scope | Where to copy |
|-------|---------------|
| User-global | \`~/.copilot/skills/autonomous-copilot/\` |
| Per-project | \`<project>/.github/skills/autonomous-copilot/\` |
| Custom | configure \`chat.skillsLocations\` |

All three paths are official Copilot Agent Skills discovery locations (per VS Code Copilot docs).

## Local verification (sandbox)
Created a Node CLI sandbox with intentional issues seeded across all 8 dimensions (hardcoded secrets, empty catch, sync I/O in loop, mega-function, missing tests, empty README, no \`--help\`, etc.) and ran the skill via \`/autonomous-skill\` in Copilot Agent mode:

- 4 sprint commits with proper conventional-commit prefixes (\`fix:\`, \`refactor:\`, \`docs:\`, \`test:\`)
- jest 7/7 tests passing
- Zero residual antipatterns (grep'd post-run)
- All 8 dimensions audited and scored (avg 7.25)
- \`session-summary.md\` generated with Feature 1-4 + PR description block
- CLI \`--help\` works end-to-end

## Known gaps (tracked for v02)
- Phase 1 question is skipped when invoked explicitly via \`/autonomous-skill\` (treats prompt as the direction) — should be conditional in SKILL.md
- Auto-invoke didn't trigger on plain "audit and improve this codebase" — \`description\` field needs more trigger keywords
- \`session-summary.md\` only shows after-scores; before/after table needs a captured baseline at Phase 2 end

## Isolation guarantees
- No existing files modified (only \`.gitignore\` adds \`/.vscode/\` for local test scaffolding)
- No script in \`scripts/\` references \`copilot/\`
- \`loop.sh\` and \`build-sprint-prompt.sh\` only load the root \`SKILL.md\` (verified via grep)
- Claude Code conductor behavior is byte-identical pre/post merge

## Test plan
- [ ] Reviewer reads \`copilot/README.md\` and follows one install path
- [ ] Reviewer opens VS Code in any small project, runs \`/autonomous-skill\` in Copilot Agent mode
- [ ] Verify the 4-phase workflow runs (or document where it deviates for v02 follow-up)

### CODE REVIEW

File: copilot/skills/autonomous-copilot/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+1. Use `grep_search` and `file_search` for the heuristic patterns
+   (e.g. `test_coverage` → look for test directories and test files;
+    `security` → grep for `password`, `secret`, `api_key`, `eval(`, `exec(`).
+2. Use `read_file` on a representative sample (don't read the whole codebase —
+   just enough to ground the score).
+3. Assign a 0-10 integer. Be honest. Inflated scores defeat the loop.
+4. Append to the `manage_todo_list` scoreboard:
+   `[test_coverage] 3/10 — only 2 test files for 47 source files, no CI runner`
+5. Save to `memory` under key `scoreboard` as `{dimension: {score, evidence}}`.
+
```
[the remark this answers is not in the document]
Comment (PengyuChen01) [reply copilot/skills/autonomous-copilot/SKILL.md:61]: Good catch — fixed in 14046cb. Renamed the Phase 2 field from `score` to `before` so Phase 3 can append `.after` without clobbering the baseline; Phase 4 now reads both fields off the same per-dimension object.

File: copilot/skills/autonomous-copilot/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+while any dimension scores < 7:
+  pick      = dimension with lowest score (ties broken by user's direction, then by phase 2 order)
+  files     = at most 5 files to touch for this fix
+  for each file in files:
+    edit
+    get_errors(file)            # halt this dimension if errors appear and aren't fixable in-place
+  run project test/lint command  # if one exists
+  rescore   = honest 0-10 after the fix
+  print     `[<dimension>] <before> → <after> ✓ (<one-line summary>)`
+  memory.update(scoreboard[<dimension>] = rescore, files_touched[<dimension>] = files)
```
[the remark this answers is not in the document]
Comment (PengyuChen01) [reply copilot/skills/autonomous-copilot/SKILL.md]: Fixed in 14046cb, same commit as the scoreboard-overwrite thread. Phase 3's loop body now opens with `before = memory[scoreboard][<dimension>].before`, so the `print` and the `if rescore <= before` check both have a real source.

File: copilot/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+cp -r copilot/skills/autonomous-copilot ~/.copilot/skills/
+```
+
+Copilot auto-discovers it across every workspace. Nothing to commit.
+
+### Option 2 — Project-local (commit + share with team)
+
+```bash
+mkdir -p <your-project>/.github/skills
+cp -r copilot/skills/autonomous-copilot <your-project>/.github/skills/
```
[the remark this answers is not in the document]
Comment (PengyuChen01) [reply copilot/README.md]: Fixed in 14046cb. Rewrote the project-local install block to `cd` into the target project first, so the bash snippet now contains only plain paths — no angle-bracket placeholders for the shell to parse as stdin/stdout redirection.
